src/rewrite.rs

```diff
@@ -129,12 +129,13 @@ impl Wizer {
                     encoder.section(&globals);
                 }
 
-                // Remove the initialization function's export and perform any
-                // requested renames.
+                // Remove exports for the wizer initialization
+                // function and WASI reactor _initialize function,
+                // then perform any requested renames.
                 s if s.id == SectionId::Export.into() => {
                     let mut exports = wasm_encoder::ExportSection::new();
                     for export in module.exports(cx) {
-                        if export.field == self.init_func {
+                        if export.field == self.init_func || export.field == "_initialize" {
```

Comment:
If there was a non-WASI reactor export named `_initialize` then this will remove it, even if we didn't already call its function (it might even be exporting a global or something like that). Since we check for the case where `_initialize` doesn't match the WASI reactor ABI and ignore it if it doesn't match during instantiation phase, that means we need to handle the ignored case later on, here, in the rewrite phase.

We can do this by passing a `bool` through to here that represents whether we called the WASI reactor `_initialize` function, and only remove the `_initialize` export if we called it already.